File: MiniDOMBase/DMD/Anagrafica/Fonti/CFontiCursor.cs
using System;
using System.Collections.Generic;
using DMD;
using DMD.Databases;
using DMD.XML;
using static minidom.Sistema;
using minidom.repositories;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Cursore sulla tabella delle fonti
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CFontiCursor 
            : minidom.Databases.DBObjectCursor<CFonte>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorStringField m_Tipo = new DBCursorStringField("Tipo");
            private DBCursorField<DateTime> m_DataInizio = new DBCursorField<DateTime>("DataInizio");
            private DBCursorField<DateTime> m_DataFine = new DBCursorField<DateTime>("DataFine");
            private DBCursorField<bool> m_Attiva = new DBCursorField<bool>("Attiva");
            private DBCursorStringField m_IconURL = new DBCursorStringField("IconURL");
            private DBCursorStringField m_IDCampagna = new DBCursorStringField("IDCampagna");
            private DBCursorStringField m_IDAnnuncio = new DBCursorStringField("IDAnnuncio");
            private DBCursorStringField m_IDKeyWord = new DBCursorStringField("IDKeyWord");
            private DBCursorStringField m_Siti = new DBCursorStringField("Siti");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");
            private bool m_OnlyValid;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CFontiCursor()
            {
                m_OnlyValid = false;
            }

            /// <summary>
            /// Nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// Tipo
            /// </summary>
            public DBCursorStringField Tipo
            {
                get
                {
                    return m_Tipo;
                }
            }

            /// <summary>
            /// IconURL
            /// </summary>
            public DBCursorStringField IconURL
            {
                get
                {
                    return m_IconURL;
                }
            }

            /// <summary>
            /// DataInizio
            /// </summary>
            public DBCursorField<DateTime> DataInizio
            {
                get
                {
                    return m_DataInizio;
                }
            }

            /// <summary>
            /// DataFine
            /// </summary>
            public DBCursorField<DateTime> DataFine
            {
                get
                {
                    return m_DataFine;
                }
            }

            /// <summary>
            /// Attiva
            /// </summary>
            public DBCursorField<bool> Attiva
            {
                get
                {
                    return m_Attiva;
                }
            }

            /// <summary>
            /// IDCampagna
            /// </summary>
            public DBCursorStringField IDCampagna
            {
                get
                {
                    return m_IDCampagna;
                }
            }

            /// <summary>
            /// IDAnnuncio
            /// </summary>
            public DBCursorStringField IDAnnuncio
            {
                get
                {
                    return m_IDAnnuncio;
                }
            }

            /// <summary>
            /// IDKeyWord
            /// </summary>
            public DBCursorStringField IDKeyWord
            {
                get
                {
                    return m_IDKeyWord;
                }
            }

            /// <summary>
            /// Siti
            /// </summary>
            public DBCursorStringField Siti
            {
                get
                {
                    return m_Siti;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<int> Flags
            {
                get
                {
                    return this.m_Flags;
                }
            }

            /// <summary>
            /// Restituisce o imposta un valore booleano che indica se il cursore deve restituire solo gli oggetti validi alla data odierna
            /// </summary>
            public bool OnlyValid
            {
                get
                {
                    return m_OnlyValid;
                }

                set
                {
                    m_OnlyValid = value;
                }
            }

            /// <summary>
            /// Aggiunge i vincoli aggiontivi 
            /// </summary>
            /// <returns></returns>
            protected override DBCursorFieldBase GetWherePartLimit()
            {
                var ret = base.GetWherePartLimit();
                if (this.OnlyValid)
                {
                    var oggi = DMD.DateUtils.Now();
                    ret *= this.Field("DataInizio").IsNull() + this.Field("DataInizio").LE(oggi);
                    ret *= this.Field("DataFine").IsNull() + this.Field("DataFine").GT(oggi);
                }
                return ret;
            }


       

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.Fonti; //.Module;
            }

            //public override string GetTableName()
            //{
            //    return "tbl_FontiContatto";
            //}

            /// <summary>
            /// Serializzazione xml
            /// </summary>
            /// <param name="writer"></param>
            protected override void XMLSerialize(XMLWriter writer)
            {
                writer.WriteAttribute("OnlyValid", m_OnlyValid);
                base.XMLSerialize(writer);
            }

            /// <summary>
            /// Deserializzazione xml
            /// </summary>
            /// <param name="fieldName"></param>
            /// <param name="fieldValue"></param>
            protected override void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "OnlyValid":
                        {
                            m_OnlyValid = DMD.XML.Utils.Serializer.DeserializeBoolean(fieldValue) == true;
                            break;
                        }

                    default:
                        {
                            base.SetFieldInternal(fieldName, fieldValue);
                            break;
                        }
                }
            }
        }
    }
}

File: MiniDOMBase/DMD/Anagrafica/Fonti/CFontiCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
 
    ''' <summary>
    ''' Cursore sulla tabella delle fonti
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CFontiCursor
        Inherits DBObjectCursor(Of CFonte)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Tipo As New CCursorFieldObj(Of String)("Tipo")
        Private m_DataInizio As New CCursorField(Of Date)("DataInizio")
        Private m_DataFine As New CCursorField(Of Date)("DataFine")
        Private m_Attiva As New CCursorField(Of Boolean)("Attiva")
        Private m_IconURL As New CCursorFieldObj(Of String)("IconURL")
        Private m_IDCampagna As New CCursorFieldObj(Of String)("IDCampagna")
        Private m_IDAnnuncio As New CCursorFieldObj(Of String)("IDAnnuncio")
        Private m_IDKeyWord As New CCursorFieldObj(Of String)("IDKeyWord")
        Private m_Siti As New CCursorFieldObj(Of String)("Siti")

        Private m_OnlyValid As Boolean

        Public Sub New()
            Me.m_OnlyValid = False
        End Sub

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property Tipo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Tipo
            End Get
        End Property

        Public ReadOnly Property IconURL As CCursorFieldObj(Of String)
            Get
                Return Me.m_IconURL
            End Get
        End Property

        Public ReadOnly Property DataInizio As CCursorField(Of Date)
            Get
                Return Me.m_DataInizio
            End Get
        End Property

        Public ReadOnly Property DataFine As CCursorField(Of Date)
            Get
                Return Me.m_DataFine
            End Get
        End Property

        Public ReadOnly Property Attiva As CCursorField(Of Boolean)
            Get
                Return Me.m_Attiva
            End Get
        End Property

        Public ReadOnly Property IDCampagna As CCursorFieldObj(Of String)
            Get
                Return Me.m_IDCampagna
            End Get
        End Property

        Public ReadOnly Property IDAnnuncio As CCursorFieldObj(Of String)
            Get
                Return Me.m_IDAnnuncio
            End Get
        End Property

        Public ReadOnly Property IDKeyWord As CCursorFieldObj(Of String)
            Get
                Return Me.m_IDKeyWord
            End Get
        End Property

        Public ReadOnly Property Siti As CCursorFieldObj(Of String)
            Get
                Return Me.m_Siti
            End Get
        End Property

        Public Property OnlyValid As Boolean
            Get
                Return Me.m_OnlyValid
            End Get
            Set(value As Boolean)
                Me.m_OnlyValid = value
            End Set
        End Property

        Public Overrides Function GetWherePart() As String
            Dim wherePart As String = MyBase.GetWherePart()
            If (Me.OnlyValid) Then
                wherePart = Strings.Combine(wherePart, "(([DataInizio] Is Null) Or ([DataInizio]<=" & DBUtils.DBDate(DateUtils.ToDay) & "))", " AND ")
                wherePart = Strings.Combine(wherePart, "(([DataFine] Is Null) Or ([DataFine]>=" & DBUtils.DBDate(DateUtils.ToDay) & "))", " AND ")
            End If
            Return wherePart
        End Function

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return APPConn
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Fonti.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_FontiContatto"
        End Function

        Protected Overrides Sub XMLSerialize(writer As XML.XMLWriter)
            writer.WriteAttribute("OnlyValid", Me.m_OnlyValid)
            MyBase.XMLSerialize(writer)
        End Sub

        Protected Overrides Sub SetFieldInternal(fieldName As String, fieldValue As Object)
            Select Case fieldName
                Case "OnlyValid" : Me.m_OnlyValid = XML.Utils.Serializer.DeserializeBoolean(fieldValue)
                Case Else : MyBase.SetFieldInternal(fieldName, fieldValue)
            End Select
        End Sub
    End Class


End Class